src/workflow/confirm.c

```diff
@@ -54,6 +56,55 @@ bool workflow_confirm_with_timeout(
     return _result;
 }
 
+static bool _have_result = false;
+
+static void _confirm_async(component_t* component)
+{
+    (void)component;
+    _result = true;
+    _have_result = true;
+    ui_screen_stack_pop();
+}
+
+static void _reject_async(component_t* component)
+{
+    (void)component;
+    _result = false;
+    _have_result = true;
+    ui_screen_stack_pop();
+}
+
+enum _confirm_state {
+    CONFIRM_IDLE,
+    CONFIRM_WAIT,
+} _confirm_state = CONFIRM_IDLE;
+
+enum ready workflow_confirm_async(
+    const char* title,
+    const char* body,
+    bool accept_only,
+    bool* result)
+{
+    switch (_confirm_state) {
+    case CONFIRM_IDLE:
+        _result = false;
+        ui_screen_stack_push(
+            confirm_create(title, body, false, _confirm_async, accept_only ? NULL : _reject_async));
+        _confirm_state = CONFIRM_WAIT;
+        /* FALLTHRU */
+    case CONFIRM_WAIT:
+        if (!_have_result) {
+            return NotReady;
+        }
+        _have_result = false;
+        _confirm_state = CONFIRM_IDLE;
+        *result = _result;
+        return Ready;
+    default:
+        Abort("Internal error");
```

Comment:
it's good to prefix abort messages with some context so it can be tracked down more easily (`workflow_confirm_async: ...`)

Reply:
ack